### ISSUE TITLE
Angular demo broken - need upcoming v8.0.0 release

### ISSUE THREAD

**Issue by AnalogJ** [gridstack/gridstack.js#2276]
## Subject of the issue
I cloned the latest master locally and tried to startup the gridstack.js angular demo. Got a ton of errors, which I'll include below. 

## Your environment
* version of gridstack.js - [`d99a9f5cb872dac8d1c102e724cdc3d2750a89e4`](https://github.com/gridstack/gridstack.js/commit/d99a9f5cb872dac8d1c102e724cdc3d2750a89e4)
* which browser/OS - macOS 12.6 

## Steps to reproduce

```
git clone git@github.com:gridstack/gridstack.js.git
cd gridstack.js/demo/angular
yarn install
ng serve

# also failed with `npm run-script start`
```

## Expected behavior

The application should start successfully and be available on localhost:4200

---

```cm

# yarn install
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 8.90s.

# ng serve
✔ Browser application bundle generation complete.

Initial Chunk Files | Names         |  Raw Size
vendor.js           | vendor        |   1.89 MB |
polyfills.js        | polyfills     | 316.96 kB |
styles.js           | styles        | 215.90 kB |
main.js             | main          |  86.53 kB |
runtime.js          | runtime       |   6.52 kB |

                    | Initial Total |   2.50 MB

Build at: 2023-04-23T22:49:20.833Z - Hash: 0964e799029af623 - Time: 4601ms

Warning: /Users/jason/repos/tmp/gridstack.js/demo/angular/src/app/app.module.ts depends on 'gridstack'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies



./src/styles.css - Error: Module build failed (from ./node_modules/css-loader/dist/cjs.js):
Error: Can't resolve '../../../dist/gridstack-extra.css' in '/Users/jason/repos/tmp/gridstack.js/demo/angular/src'

./src/styles.css?ngGlobalStyle - Error: Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Module build failed (from ./node_modules/css-loader/dist/cjs.js):
Error: Can't resolve '../../../dist/gridstack-extra.css' in '/Users/jason/repos/tmp/gridstack.js/demo/angular/src'

Error: src/app/app.component.ts:54:26 - error TS2322: Type '{ x: number; y: number; w: number; h: number; subGridOpts: { acceptWidgets?: string | boolean | ((element: Element) => boolean) | undefined; addRemoveCB?: AddRemoveFcn | undefined; ... 40 more ...; subGridDynamic?: boolean | undefined; }; }' is not assignable to type 'NgGridStackWidget'.
  Object literal may only specify known properties, and 'subGridOpts' does not exist in type 'NgGridStackWidget'.

54     {x:1, y:0, w:4, h:4, subGridOpts: {children: this.sub1, id:'sub1_grid', class: 'sub1', ...this.subOptions}},
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: src/app/app.component.ts:55:26 - error TS2322: Type '{ x: number; y: number; w: number; h: number; subGridOpts: { acceptWidgets?: string | boolean | ((element: Element) => boolean) | undefined; addRemoveCB?: AddRemoveFcn | undefined; ... 40 more ...; subGridDynamic?: boolean | undefined; }; }' is not assignable to type 'NgGridStackWidget'.
  Object literal may only specify known properties, and 'subGridOpts' does not exist in type 'NgGridStackWidget'.

55     {x:5, y:0, w:3, h:4, subGridOpts: {children: this.sub2, id:'sub2_grid', class: 'sub2', ...this.subOptions}},
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: src/app/app.component.ts:71:25 - error TS2339: Property 'subGridOpts' does not exist on type 'NgGridStackWidget'.

71       if (!w.type && !w.subGridOpts) w.content = `item ${ids}`;
                           ~~~~~~~~~~~


Error: src/app/app.component.ts:96:28 - error TS2339: Property 'addRemoveCB' does not exist on type 'typeof GridStack'.

96     if (val < 3) GridStack.addRemoveCB = undefined;
                              ~~~~~~~~~~~


Error: src/app/app.component.ts:97:20 - error TS2339: Property 'addRemoveCB' does not exist on type 'typeof GridStack'.

97     else GridStack.addRemoveCB = gsCreateNgComponents;
                      ~~~~~~~~~~~


Error: src/app/app.component.ts:145:39 - error TS2339: Property 'engine' does not exist on type 'GridStackOptions | GridStack'.
  Property 'engine' does not exist on type 'GridStackOptions'.

145     if (node?.subGrid && node.subGrid.engine.nodes.length) {
                                          ~~~~~~


Error: src/app/app.component.ts:146:7 - error TS2322: Type 'GridStackOptions | GridStack' is not assignable to type 'GridStack | undefined'.
  Type 'GridStackOptions' is missing the following properties from type 'GridStack': el, engine, opts, addWidget, and 34 more.

146       grid = node.subGrid;
          ~~~~


Error: src/app/app.component.ts:147:14 - error TS18048: 'grid' is possibly 'undefined'.

147       node = grid.engine.nodes[0];
                 ~~~~


Error: src/app/app.component.ts:149:15 - error TS18048: 'grid' is possibly 'undefined'.

149     if (node) grid.removeWidget(node.el!);
                  ~~~~


Error: src/app/app.module.ts:40:15 - error TS2339: Property 'addRemoveCB' does not exist on type 'typeof GridStack'.

40     GridStack.addRemoveCB = gsCreateNgComponents; // DONE in case switcher onShow() as well
                 ~~~~~~~~~~~


Error: src/app/app.module.ts:41:15 - error TS2339: Property 'saveCB' does not exist on type 'typeof GridStack'.

41     GridStack.saveCB = gsSaveAdditionNgInfo;
                 ~~~~~~


Error: src/app/gridstack.component.ts:219:14 - error TS2339: Property 'subGridOpts' does not exist on type 'GridStackOptions | NgGridStackWidget'.
  Property 'subGridOpts' does not exist on type 'GridStackOptions'.

219       if (!w.subGridOpts && type && GridstackComponent.selectorToType[type]) {
                 ~~~~~~~~~~~




** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **


✖ Failed to compile.

```

**Comment by dbvcode** [comment]
Issue confirmed with https://github.com/gridstack/gridstack.js/commit/d99a9f5cb872dac8d1c102e724cdc3d2750a89e4

**Comment by adumesny** [comment]
@AnalogJ @dbvcode I was working on a newer angular wrapper over the weekend, but a new v8.0 will need to be published for it... (and doc update).

for now you can either use the older published v7.2.3 version or 'yarn build' the lib itself and copy the /dist content to the angular node_modules location (what I do) to get the demo (and I assume your code) working for the future release...

funny thing is I never hear when people are using the lib or wrapper, until I break something :)
I'm using the ng wrapper for work, so there will be more changes/improvements as I find them...

**Comment by adumesny** [comment]
fixed
